### PR TITLE
[FLINK-8703][tests] Port CancelingTestBase to MiniClusterResource

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

Ports the `CancelingTestBase` to use `MiniClusterResource`.

## Verifying this change

Run `MapCancelingTestBase` with `flip6` profile enabled/disabled.

### CODE REVIEW

**Comment by tillrohrmann** [comment]
Interesting, there is a aborted test run with a core dump.

File: flink-tests/src/test/java/org/apache/flink/test/cancelling/JoinCancelingITCase.java [hunk 1]
```diff
@@ -30,19 +30,18 @@
 import org.apache.flink.test.util.InfiniteIntegerTupleInputFormat;
 import org.apache.flink.test.util.UniformIntTupleGeneratorInputFormat;
 
+import org.junit.Ignore;
+import org.junit.Test;
+
 /**
  * Test job cancellation from within a JoinFunction.
  */
+@Ignore("Takes too long.")
```
Comment (tillrohrmann) [inline flink-tests/src/test/java/org/apache/flink/test/cancelling/JoinCancelingITCase.java]: Do you know what's taking so long?

File: flink-tests/src/test/java/org/apache/flink/test/cancelling/CancelingTestBase.java [hunk 2]
```diff
@@ -18,160 +18,111 @@
...
 	// --------------------------------------------------------------------------------------------
 
-	protected LocalFlinkMiniCluster executor;
-
-	protected int taskManagerNumSlots = DEFAULT_TASK_MANAGER_NUM_SLOTS;
+	@ClassRule
+	public static final MiniClusterResource CLUSTER = new MiniClusterResource(
+		new MiniClusterResource.MiniClusterResourceConfiguration(
+			getConfiguration(),
+			2,
```
Comment (tillrohrmann) [inline flink-tests/src/test/java/org/apache/flink/test/cancelling/CancelingTestBase.java]: Why do we start a mini cluster with 2 TMs and 4 slots per TM? Wouldn't a single TM be sufficient?

Comment (zentol) [reply flink-tests/src/test/java/org/apache/flink/test/cancelling/CancelingTestBase.java]: Possibly, i just copied the existing setup.

File: flink-tests/src/test/java/org/apache/flink/test/cancelling/JoinCancelingITCase.java (hunk 1 above)
Comment (zentol) [reply flink-tests/src/test/java/org/apache/flink/test/cancelling/JoinCancelingITCase.java]: No, I also didn't check how long it actually takes.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**